File: Trees/Traversal/Traversal.js
class Node{
    constructor(element){
       this.data=element
       this.left=null
       this.right=null
    }
}

function preorderTraversal(node){
    //return if node is null
    if(node==null){
        return
    }
    //Print the value of Current Node
    console.log(node.data)
    //make recursive call to the left subtree
    preorderTraversal(node.left)
    //make recursive call to right subtree
    preorderTraversal(node.right)
}

function inorderTraversal(node){
    //return if node is null
    if(node==null){
        return
    }

    //make recursive call to the left subtree
    inorderTraversal(node.left)
    //Print the value of Current Node
    console.log(node.data)
    //make recursive call to right subtree
    inorderTraversal(node.right)
}

function postorderTraversal(node){
    //return if node is null
    if(node==null){
        return
    }
    
    //make recursive call to the left subtree
    postorderTraversal(node.left)
    //make recursive call to right subtree
    postorderTraversal(node.right)
    //Print the value of Current Node
    console.log(node.data)
}

let root=new Node(1)
root.left=new Node(2)
root.right=new Node(3)
root.left.left=new Node(4)
root.left.right=new Node(5)

console.log("PreOrder traversal")
preorderTraversal(root)

console.log("InOrder traversal")
inorderTraversal(root)

console.log("PostOrder traversal")
postorderTraversal(root)

File: Trees/Traversal/Traversal.py
class Node:
    def __init__(self,element):
        self.data=element
        self.left=None
        self.right=None

def preorderTraversal(root):
    #Enter if Node is not Null
    if root:
        #Print The Value of The Node
        print(str(root.data)+"->",end='')
        #Make Recursive Call to the Left Subtree
        preorderTraversal(root.left)
        #Make Recursive Call to the Right Subtree
        preorderTraversal(root.right)

def inorderTraversal(root):
    #Enter if Node is not Null
    if root:
        #Make Recursive Call to the Left Subtree
        inorderTraversal(root.left)
        #Print The Value of The Node
        print(str(root.data)+"->",end='')
        #Make Recursive Call to the Right Subtree
        inorderTraversal(root.right)

def postorderTraversal(root):
    #Enter if Node is not Null
    if root:
        #Make Recursive Call to the Left Subtree
        postorderTraversal(root.left)
        #Make Recursive Call to the Right Subtree
        postorderTraversal(root.right)        
        #Print The Value of The Node
        print(str(root.data)+"->",end='')
        



root=Node(1)
root.left=Node(2)
root.right=Node(3)
root.left.left=Node(4)
root.left.right=Node(5)

print("PreOrder Traversal ")
preorderTraversal(root)

print("\nInOrder Traversal ")
inorderTraversal(root)

print("\nPostOrder Traversal ")
postorderTraversal(root)